src/components/TodoAppContent/TodoAppContent.tsx

```diff
@@ -0,0 +1,237 @@
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useRef,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  editTodo,
+} from '../../api/todos';
+import { useAuth } from '../../Context/Context';
+import { TodoItem } from '../TodoItem';
+import { Filter } from '../Filter';
+import { filteredData } from '../../helpers/filteredData';
+import { FilterBy, ErrorMessage, EditField } from '../../types/types';
+import { Error } from '../Error';
+
+export const TodoAppContent: React.FC = () => {
+  const userId = useAuth();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoField, setNewTodoField] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todosCount, setTodosCount] = useState(0);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadTodosIds, setLoadTodosIds] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [disabledClear, setDisabledClear] = useState(true);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const currentTodos = filteredData<Todo>(todos, filterBy);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const countOfNotCompleted = todos
+    .filter(({ completed }) => !completed);
+
+  const countOfCompleted = todos.filter(({ completed }) => completed);
+
+  useEffect(() => {
+    if (userId) {
+      getTodos(userId)
+        .then((resp) => {
+          setTodos(resp);
+        })
+        .catch(() => setErrorMessage(ErrorMessage.Load));
+    }
+  }, [userId]);
+
+  useEffect(() => {
+    setTodosCount(countOfNotCompleted.length);
+  }, [todos]);
+
+  useEffect(() => {
+    if (!isAdding) {
+      inputRef.current?.focus();
+    }
+  }, [isAdding]);
+
+  useMemo(() => {
+    setDisabledClear(!todos.some(({ completed }) => completed));
+  }, [todos]);
+
+  const handleAddTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (userId) {
+      if (!newTodoField.trim()) {
+        setErrorMessage(ErrorMessage.NotBeEmpty);
+
+        return;
+      }
+
+      const body = {
+        id: 0,
+        title: newTodoField.replace(/\s+/g, ' ').trim(),
+        userId,
+        completed: false,
+      };
+
+      setIsAdding(true);
+      setTempTodo(body);
+
+      addTodo(body)
+        .then(resp => {
+          setTodos(prev => [...prev, resp]);
+          setNewTodoField('');
+        })
+        .catch(() => {
+          setErrorMessage(ErrorMessage.Add);
+        })
+        .finally(() => {
+          setTempTodo(null);
+          setIsAdding(false);
+        });
+    }
+  };
+
+  const handleDeleteTodo = (idToDel: number) => {
```

Comment:
It's not good idea to reduce variables names, better to call this parameter idToDelete